daemon/cdi.go

```diff
@@ -69,11 +68,23 @@ func createCDICache(cdiSpecDirs ...string) (*cdi.Cache, error) {
 		return nil, fmt.Errorf("no CDI specification directories specified")
 	}
 
-	cache, err := cdi.NewCache(cdi.WithSpecDirs(cdiSpecDirs...))
+	err := cdi.Configure(cdi.WithSpecDirs(cdiSpecDirs...))
 	if err != nil {
 		return nil, fmt.Errorf("CDI registry initialization failure: %w", err)
 	}
 
+	cache := cdi.GetDefaultCache()
```

Comment:
So moby and buildkit are now using the same default cache instance instead of creating a new one which seems to be the same for containerd: https://github.com/containerd/containerd/blob/f6724ac31904c4d41d6794b4df130565829b5621/cmd/ctr/commands/run/run_unix.go#L482 so we should be good.

When containerd runs as a service it also uses the default cache: https://github.com/containerd/containerd/blob/f6724ac31904c4d41d6794b4df130565829b5621/internal/cri/server/service_linux.go#L105. But no issue here as it's pretty much a standalone product.

Reply:
Good one; @AkihiroSuda @fuweid  any idea why CRI doesn't auto-refresh? Perhaps it was not an option yet at the time (I see it was added 3 Years ago)

Looks like it was originally added in https://github.com/containerd/containerd/commit/8ed910c46ae146069a8f85dc10a8eb4de1f84351 ;

- https://github.com/containerd/containerd/pull/7419

Reply:
Since `ctr` is only concerned with the state of the cache AT invocation, we disable autorefresh when injecting devices in the CTR command. It is my understanding that this should not be affecting the behaviour of the cache for the long-running contianerd service, since the two default cache instances created are in fact different.